core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -300,6 +304,7 @@ public void onTopologyUpdate(final CacheTopology cacheTopology, final boolean is
       // Ensures writes to the data container use the right consistent hash
       // No need for a try/finally block, since it's just an assignment
       stateTransferLock.acquireExclusiveTopologyLock();
+      triangleOrderManager.updateCacheTopology(cacheTopology);
```

Comment:
the cache topology is only used to assign the sequence number to the backup commands. It doesn't affect the deliver order.